pithos/pithos.py

```diff
@@ -415,6 +433,7 @@ def set_audio_quality(self, *ignore):
         self.worker_run('set_audio_quality', (self.settings.get_string('audio-quality'),))
 
     def pandora_connect(self, *ignore, message="Logging in...", callback=None):
+        self.connected = False
```

Comment:
This might be helpful for you: https://python-gtk-3-tutorial.readthedocs.org/en/latest/objects.html

A property seems a bit simpler than a signal for this.

pandora.py:
```python
class Pandora(GObject.GObject):
    connected = GObject.Property(type=bool, default=False)
    ...
    def connect(...):
        ...
        self.connected = True
```

pithos.py:
```python
pandora.conect('notify::connected', ...)
```